plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
+		{
+			name: "invalid ipv4",
+			config: `{
+				"ip": "192.0.2.1.1",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip":"not an ip address: 192.0.2.1.1"}}`,
+		},
+		{
+			name: "valid ipv6",
+			config: `{
+				"ip": "2001:db8::1",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ipv6",
+			config: `{
+				"ip": "2001:db8::1::1",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip":"not an ip address: 2001:db8::1::1"}}`,
+		},
+		{
+			name: "valid ip_or_cidr",
+			config: `{
+				"ip_or_cidr": "192.0.2.0/24",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ip_or_cidr with too big mask",
+			config: `{
+				"ip_or_cidr": "192.0.2.0/244",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip_or_cidr":"invalid ip or cidr range: '192.0.2.0/244'"}}`,
+		},
+		{
+			name: "valid cidr_v4",
+			config: `{
+				"cidr_v4": "192.0.2.0/24",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid cidr_v4 with too big mask",
+			config: `{
+				"cidr_v4": "192.0.2.0/244",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"cidr_v4":"invalid ipv4 cidr range: '192.0.2.0/244'"}}`,
+		},
+		{
+			name: "valid port",
+			config: `{
+				"port": 80,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid port as string",
+			config: `{
+				"port": "80",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"expected an integer"}}`,
+		},
+		{
+			name: "invalid too big port",
+			config: `{
+				"port": 9999999999,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"value should be between 0 and 65535"}}`,
+		},
+		{
+			name: "invalid negative port",
+			config: `{
+				"port": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"value should be between 0 and 65535"}}`,
+		},
+		{
+			name: "valid url",
+			config: `{
+				"url": "https://www.konghq.com",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "valid header_name",
+			config: `{
+				"header_name": "x-header",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid character in header_name",
+			config: `{
+				"header_name": "!header",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{"header_name":"bad header name '!header', ` +
+				`allowed characters are A-Z, a-z, 0-9, '_', and '-'"}}`,
+		},
+		{
+			name: "valid timeout",
+			config: `{
+				"timeout": 10,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid negative timeout",
+			config: `{
+				"timeout": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"timeout":"value should be between 0 and 2147483646"}}`,
+		},
+		{
+			name: "invalid too big timeout",
+			config: `{
+				"timeout": 2147483647,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"timeout":"value should be between 0 and 2147483646"}}`,
+		},
```

Comment:
We should hit the hard edges too; 0 and 2147483646.

Reply:
https://github.com/Kong/goks/pull/14/commits/9d5314df4c1e19a42edfbc527b7f4793af9e7f39